detox/src/artifacts/templates/plugin/WholeTestRecorderPlugin.js

```diff
@@ -23,12 +23,13 @@ class WholeTestRecorderPlugin extends ArtifactPlugin {
     await super.onTestDone(testSummary);
 
     if (this.testRecording) {
-      await this.testRecording.stop();
+      const testRecording = this.testRecording;
+      await testRecording.stop();
```

Comment:
Similar case here, I suppose?

Reply:
If so, wouldn't calling `startSavingTestRecording` with an obsolete `testRecording` (whatever that means) could end-up in an invalid artifact?